packages/app/src/firebaseNamespaceCore.ts

```diff
@@ -234,6 +236,22 @@ export function createFirebaseNamespaceCore(
       : null;
   }
 
+  function registerVersion(library: string, version: string): void {
+    if (library.match(/\s|\//)) {
+      logger.warn(
+        `Could not register ${library}: it contains illegal characters.`
+      );
+      return;
+    }
+    registerComponent(
+      new Component(
+        `${library}-version` as Name,
```

Comment:
I'm not a fan of this cast TBH, since it defeats the purpose of the `Name` type. Maybe we shouldn't consider versions actual `Component`s and instead create a different type for them?

Reply:
I think that it might be a good idea in the long term to have separate Map in `ComponentContainer` in addition to the `Providers` Map, and this new Map just stores key/value pairs. Can be used for versions now, and other possible key/value pair data going forward.  But that requires a bit of modification to ComponentContainer that might not be a good idea right now as we're planning to do our first release with the migration to the component model next week right before a holiday.  Since the actual `registerVersion` API wouldn't change we can change the implementation later when the component model is in use and stable.

Reply:
Sure, sounds good.